src/main/java/com/ait/lienzo/client/core/shape/TextBoundsAndLineBreaksWrap.java

```diff
@@ -0,0 +1,130 @@
+        super(text);
+    }
+
+    public TextBoundsAndLineBreaksWrap(final Text text,
+                                       final BoundingBox wrapBoundaries) {
+        super(text, wrapBoundaries);
+    }
+
+    public TextBoundsAndLineBreaksWrap(final Supplier<String> textSupplier,
+                                       final Supplier<Double> fontSizeSupplier,
+                                       final Supplier<String> fontStyleSupplier,
+                                       final Supplier<String> fontFamilySupplier,
+                                       final Supplier<TextUnit> textUnitSupplier,
+                                       final Supplier<TextBaseLine> textBaseLineSupplier,
+                                       final Supplier<TextAlign> textAlignSupplier,
+                                       final BoundingBox wrapBoundaries) {
+        super(textSupplier,
+              fontSizeSupplier,
+              fontStyleSupplier,
+              fontFamilySupplier,
+              textUnitSupplier,
+              textBaseLineSupplier,
+              textAlignSupplier,
+              wrapBoundaries);
+    }
+
+    @Override
+    public void drawString(final Context2D context,
+                           final Attributes attr,
+                           final IDrawString drawCommand) {
+        final BoundingBox wrapBoundaries = getWrapBoundaries();
+        final String[] textLines = attr.getText().split("\\r?\\n");
+        if (textLines.length < 1) {
+            return;
+        }
+
+        final ArrayList<String> lines = new ArrayList<>();
+        for (String line : textLines) {
+            String[] words = line.split("\\s");
+            if (words.length < 1) {
+                lines.add("");
+                continue;
+            }
+            final StringBuilder nextLine = new StringBuilder(words[0]);
+
+            for (int i = 1; i < words.length; i++) {
+                if (getBoundingBoxForString(nextLine + " " + words[i]).getWidth() <= wrapBoundaries.getWidth()) {
+                    nextLine.append(" ").append(words[i]);
+                } else {
+                    lines.add(nextLine.toString());
+                    nextLine.setLength(words[i].length());
+                    nextLine.replace(0,
+                                     words[i].length(),
+                                     words[i]);
+                }
+            }
+            lines.add(nextLine.toString());
+        }
+
+        double xOffset = 0;
+
+        switch (textAlignSupplier.get()) {
+            case START:
+            case LEFT:
+                xOffset = 0;
+                break;
+
+            case CENTER:
+                xOffset = wrapBoundaries.getWidth() / 2;
+                break;
+
+            case END:
+            case RIGHT:
+                xOffset = wrapBoundaries.getWidth();
+                break;
+        }
+
+        double lineHeight = TextUtils.getBoundingBox("Mg", fontSizeSupplier.get(),
```

Comment:
I know @Christopher-Chianelli commented about the inability to easily write another ```ITextWrapper``` outside of Lienzo due to the accessibility of some classes/interfaces/methods etc... so I wonder whether it'll be a good idea to make the utility methods on ```TextUtils``` public instead of package-protected. WDYT?

Reply:
@manstis Good idea, although I don't see how many applicable some of the `TextUtils` methods are. I know `getBoundingBox(...)` and `padString(...)` will probably be useful, don't know about the others; I don't see anything that can cause harm if we make them all public, so I think I will go with that.